lib/Driver/ToolChains.cpp

```diff
@@ -1211,12 +1211,33 @@ toolchains::GenericUnix::constructInvocation(const LinkJobAction &job,
     Arguments.push_back(context.Args.MakeArgString(LibProfile));
   }
 
-  // FIXME: We probably shouldn't be adding an rpath here unless we know ahead
-  // of time the standard library won't be copied.
-  Arguments.push_back("-Xlinker");
-  Arguments.push_back("-rpath");
-  Arguments.push_back("-Xlinker");
-  Arguments.push_back(context.Args.MakeArgString(RuntimeLibPath));
+  if (getTriple().isAndroid()) {
```

Comment:
Sorry for the slow turnaround on this one! I'm beginning to better understand the relationship between the `Driver` and the `ToolChain`. I'm thinking of modifying `Driver::getToolChain` such that it returns a `toolchains::Android`. That should help isolate this Android-specific logic.